### PR TITLE
feat: LSP e2e test

### PR DESCRIPTION
**Pull request by haifeng-li-at-salesforce**
### What does this PR do?
Support e2e tests for LSP client within extension test. The JS file can be opened in extension and LSP is able to diagnose the file.

Support mocha debug test for LSP server

### CODE REVIEW

File: lsp/client/testFixture/adapters-localchange-not-aware.js
Comment (ben-zhang-at-salesforce) [inline lsp/client/testFixture/adapters-localchange-not-aware.js:1]: can/should we move this fixture under test folder?

File: lsp/client/src/test/diagnostic.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import { getDocUri, activate } from './helper';
+import * as sinon from 'sinon';
+
+import { afterEach } from 'mocha';
+
+suite('Diagnostics Test Suite [Client]', () => {
+    afterEach(function () {
+        sinon.restore();
+    });
+    test('Diagnose local change not aware adapter', async () => {
```
Comment (ben-zhang-at-salesforce) [inline lsp/client/src/test/diagnostic.test.ts:18]: should we create skeleton of three tests, one cover js, one cover html, one cover graphql?

File: lsp/client/src/test/diagnostic.test.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    eLine: number,
+    eChar: number
+): vscode.Range {
+    return new vscode.Range(
+        new vscode.Position(sLine, sChar),
+        new vscode.Position(eLine, eChar)
+    );
+}
+
+async function testDianostics(
```
Comment (ben-zhang-at-salesforce) [inline lsp/client/src/test/diagnostic.test.ts:43]:
```suggestion
async function testDiangostics(
```

File: lsp/client/src/test/diagnostic.test.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    docUri: vscode.Uri,
+    expectedDiagnostics: vscode.Diagnostic[]
+) {
+    await activate(docUri);
+
+    const actualDiagnostics = vscode.languages.getDiagnostics(docUri);
+    assert.equal(actualDiagnostics.length, expectedDiagnostics.length);
+    expectedDiagnostics.forEach((expectedDiagnostic, i) => {
+        const actualDignostic = actualDiagnostics[i];
```
Comment (ben-zhang-at-salesforce) [inline lsp/client/src/test/diagnostic.test.ts:52]: minor typo: ```suggestion
        const actualDiagnostic = actualDiagnostics[i];
```

File: lsp/client/src/test/diagnostic.test.ts
Comment (ben-zhang-at-salesforce) [inline lsp/client/src/test/diagnostic.test.ts:1]: it's really great to have end to end test for get-go to prevent future breakage without knowing it.

**Review by ben-zhang-at-salesforce — APPROVED** [review]
looks good, great to e2e tests.

File: lsp/client/testFixture/adapters-localchange-not-aware.js
Comment (haifeng-li-at-salesforce) [reply lsp/client/testFixture/adapters-localchange-not-aware.js:1]: fixture folder under 'src' will be moved to 'out'

File: lsp/client/src/test/diagnostic.test.ts (hunk 1 above)
Comment (haifeng-li-at-salesforce) [reply lsp/client/src/test/diagnostic.test.ts:18]: Good suggestion

File: lsp/client/src/test/diagnostic.test.ts (hunk 2 above)
Comment (haifeng-li-at-salesforce) [reply lsp/client/src/test/diagnostic.test.ts:43]: updated

**Pull request merged**